Classes/OAuthClient.php

```diff
@@ -131,7 +131,7 @@ public function injectEntityManager(EntityManagerInterface $entityManager): void
      *
      * @return string For example, "Github", "oidc", ...
      */
-    abstract public function getServiceType(): string;
```

Comment:
This would be a breaking change, since all subclasses need to be adjusted.
Instead we might be able to fix this in `Flownative\OAuth2\Client\Controller\OAuthController::detectServiceTypes()` by changing the lines
```php
            $serviceType = call_user_func_array([$serviceTypeClassName, 'getServiceType'], []);
            $serviceTypes[$serviceType] = $serviceTypeClassName;
```
to
```php
            /** @var OAuthClient $client */
            $client = $objectManager->get($serviceTypeClassName);
            $serviceTypes[$client->getServiceType()] = $serviceTypeClassName;
```

(untested)

Reply:
@svparijs You marked this as resolved without a further comment, but why? I think this is a valid concern.

A problem with the sugestion is the required constructor argument for the clients, though.…